File: problems/77. Combinations/main.go
func combine(n int, k int) [][]int {
	var result [][]int

	var helper func(start int, path []int)
	helper = func(start int, path []int) {
		left := n - start
		if len(path)+left < k {
			return
		}
		if len(path) == k {
			uniqueCombination := make([]int, k)
			copy(uniqueCombination, path)
			result = append(result, uniqueCombination)
			return
		}

		for i := start; i < n; i++ {
			path = append(path, i+1)
			helper(i+1, path)
			path = path[:len(path)-1]
		}
	}

	helper(0, []int{})
	return result
}

File: problems/77. Combinations/main.py
class Solution(object):
    def combine(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[List[int]]
        """
        result = []

        def helper(start, path):
            left = n - start
            if len(path) + left < k:
                return
            if len(path) == k:
                result.append(path[:])
                return

            for i in range(start, n):
                path.append(i + 1)
                helper(i + 1, path)
                path.pop()

        helper(0, [])
        return result
